python/ray/serve/controller.py

```diff
@@ -311,11 +315,11 @@ def get_root_url(self):
     async def shutdown(self):
         """Shuts down the serve instance completely."""
         async with self.write_lock:
-            self.deployment_state_manager.shutdown()
-            self.endpoint_state.shutdown()
+            await self.deployment_state_manager.shutdown()
+            await self.endpoint_state.shutdown()
             self.http_state.shutdown()
 
-    def deploy(
+    async def deploy(
```

Comment:
Does this require us to make the [`ServeControllerClient`'s `deploy`](https://github.com/ray-project/ray/blob/8b3451318c8bd172d59cf3d1a0b6cee266a1f642/python/ray/serve/client.py#L233) function asynchronous as well?